src/web/footer/Footer.js

```diff
@@ -0,0 +1,10 @@
+import { createElement } from '../utils/dom';
+import './footer.css';
+
+export default function Footer() {
+  const footer = createElement('footer');
+  const text = createElement('span', { textContent: 'Copyright 2025. woowacourse' });
+
+  footer.appendChild(text);
+  return footer;
+}
```

Comment:
이런 부분이 리바이가 본문에서 말한 고민되는 부분인거 같은데요. 아래처럼 작성하면 보다 읽기 쉬운 코드로 작성할 수 있을거 같아요

```suggestion
export default function Footer() {
  return /*html*/ `
    <footer>
      <span>Copyright 2025. woowacourse</span>
    </footer>
  `;
}
```

Reply:
리바이가 작성해주신 방법이랑 제가 알려드린 방법은 각각 장단점이 있는데요. 어떤 장단점이 있는지 찾아보셔도 좋을거 같네용 ~~

Reply:
좋은 주제에 대해서 예시와 함께 생각해줄 수 있도록 리뷰 해주셔서 감사해요 유조 🍀 
질문해주신 내용을 토대로 개인적으로 공부해봤어요!

# 1. createElement 방식
## 장점
- 1. 요소 생성 시 EventListener 를 바로 등록할 수 있어서 용이
- 2. textContent나 setAttribute를 사용하면 XSS 공격을 방지 가능

## 단점
- 1. 구문이 장황해질 수 있다.
- 2. 여러 요소들을 만들고 붙이는 과정에서 Dom 접근 및 조작이 더 많아지고 복잡해질 수 있음.

# 2.  템플릿 리터럴 (innerHTML) 방식
## 장점
- 1. 간결하고 직관적 : HTML을 그대로 작성할 수 있어 가독성이 좋고, 복잡한 구조도 쉽게 표현할 수 있음.
- 2. 빠른 렌더링 속도 : innerHTML은 브라우저 엔진이 내부적으로 최적화되어 있어 대량의 HTML을 빠르게 삽입 가능
- 3. 복잡한 HTML 구조 생성에 유리
## 단점
- 1. 보안 취약 -> XSS 위험 - 사용자 입력을 포함할 때는 직접적으로 사용하면 위험
- 2. addEventListener를 사용하려면, querySelector 같은 추가적인 DOM 로직 필요 
- 3. innerHTML을 사용하면 기존 요소를 완전히 초기화하므로, 기존 이벤트 리스너나 데이터가 사라질 수 있음


정리를 해보자면 저희 로또 서비스 같은 경우는 input을 받는 로직이 있어서 innerHTML보단 createElement를 사용하여 textContent 등을 사용해 XSS 위험을 방지하고, eventListener가 들어가는 동적 요소 로직이 많아서 createElement 를 사용하는게 더 효율적일 것 같다라고 개인적으로 판단하였습니다!

혹시 유조께서는 어떻게 생각하실까요?

Reply:
개인적으로는 `createElement`를 사용했을 때 tag가 중첩될수록 가독성이 너무 떨어지고 유지보수가 힘들어져서 템플릿 리터럴 방식을 사용하는 것을 선호합니다.

하지만 이건 제 의견일 뿐 정답은 없는 부분이라고 생각합니다 ㅎㅎ